op-batcher/batcher/driver.go

```diff
@@ -843,27 +843,30 @@ func (l *BatchSubmitter) sendTransaction(txdata txData, queue *txmgr.Queue[txRef
 // sendTx uses the txmgr queue to send the given transaction candidate after setting its
 // gaslimit. It will block if the txmgr queue has reached its MaxPendingTransactions limit.
 func (l *BatchSubmitter) sendTx(txdata txData, isCancel bool, candidate *txmgr.TxCandidate, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) {
-	isEOAPointer := l.inboxIsEOA.Load()
-	if isEOAPointer == nil {
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
-		defer cancel()
-		var code []byte
-		code, err := l.L1Client.CodeAt(ctx, *candidate.To, nil)
-		if err != nil {
-			l.Log.Error("CodeAt failed, assuming code exists", "err", err)
-			// assume code exist, but don't persist the result
-			isEOA := false
-			isEOAPointer = &isEOA
-		} else {
-			isEOA := len(code) == 0
-			isEOAPointer = &isEOA
-			l.inboxIsEOA.Store(isEOAPointer)
+	var isEOAPointer *bool
+	if l.RollupConfig.UseInboxContract() {
+		isEOAPointer = l.inboxIsEOA.Load()
```

Comment:
It's a bit confusing why the batch inbox could be a EOA when `RollupConfig.UseInboxContract` is set to `true`

Reply:
`UseInboxContract` just means the transaction status matters, but the actual inbox may still be an EOA.

Reply:
Yes, I understand this because I have the context. However, for the sake of code maintainability, I’d suggest using better variable names or at least adding comments for clarity.

Reply:
Done in [d73d77b](https://github.com/ethstorage/optimism/pull/122/commits/d73d77bb31e24ec789863e0946ff1054a6b9011e)